doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Scorecard Custom Test Framework
+authors:
+  - "@jemccorm"
+reviewers:
+  - TBD
+  - "@joelandford"
+  - "@zeus"
+approvers:
+  - TBD
+  - "@shurley"
+  - "@dmesser"
+creation-date: 2020-03-19
+last-updated: 2020-03-19
+status: implementable
+---
+
+# Scorecard - Custom Tests
+
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created 
+
+## Summary
+
+This enhancement proposal outlines the SDK scorecard’s design for running of custom tests, specifically tests that are outside of what Red Hat includes by default within the scorecard’s tests.   The proposed design would provide both a means to write custom tests and also convert the existing scorecard tests (basic and olm) into the custom test format.
+
+## Motivation
+
+The scorecard needs a robust means of enabling end-users of the SDK to write and execute their own operator tests.   Ideally, the custom test framework would decouple test logic from within the SDK itself, making the SDK and scorecard logic in particular more flexible and maintainable in the long run.
+
+## Goals
+
+ * Document a design for scorecard to execute end-user or custom tests.
+
+ * Document stories/steps required to deliver scorecard custom test functionality.
+
+### Non-Goals
+
+ * This is a design proposal and does not include a proof-of-concept.
+
+## Proposal
+
+This design proposal is to establish a pattern for implementing custom scorecard tests and also support the goal of migrating existing scorecard tests into the same format as the custom tests.
+
+With this proposal, the scorecard will continue to setup and teardown the Operator being tested, as well as creating Custom Resources being tested.
+
+### User Stories
+
+#### Story 1
+
+Scorecard users will be able to construct custom tests and execute them using the scorecard.
+
+#### Story 2
+
+Scorecard will ship with an example custom test that end users could use as a reference when writing their custom tests.
+
+#### Story 3
+
+Scorecard internal tests will migrate to the custom test format and therefore be externalized from the scorecard binary itself.
+
+#### Story 4
+
+Scorecard would support testing a bundle image.  
+
+#### Story 5 (optional)
+
+Scorecard would support the creation of a Kuttl test image.  This custom test would execute kuttl to perform testing enabling scorecard users a means to write tests using kuttl but have the result integrated into scorecard results.  
+
+## Design Details
+
```

Comment:
WDYT about before described the details over each smaller unit, do a technical overview of how it actually will work to make easier the understatement?  

Something such as: 

```suggestion
 A custom test is any sort of test logic required, written in any programming language that satisfies the scorecard test requirements for output format and execution.
 
In this way, the Operator developer is responsable to provide an image with all requirements and instruction to executed their tests. Scorecard requires that some standards are followed such as the configuration that will be used by the feature and the format of its output logs which will be parsed and integrated to the Scorecard results in order to work.  

The scorecard usage will be like:

`operator-sdk scorecard --bundle-image=quay.io/myorg/mytest`

In this way, scorecard basically will run the project via its bundle and the custom tests image provided and then, get its output logs to be used in its own results. 

**NOTES** See that SDK already has a feature to create the [bundle](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/olm-catalog/operator-bundles.md). 
```

PS: If I properly understood too :-P

Reply:
I updated that command a bit to make it clearer (I hope) :smile: